src/utl/src/ScopedTemporaryFile.cpp

```diff
@@ -33,4 +36,70 @@ ScopedTemporaryFile::~ScopedTemporaryFile()
   }
 }
 
+StreamHandler::StreamHandler(const char* filename, bool binary)
+    : filename_(filename)
+{
+  std::string tmp_filename = filename_ + ".tmp";
+  if (fs::exists(tmp_filename)) {
+    fs::remove(tmp_filename);
+  }
+  os_.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+  try {
+    if (binary) {
+      os_.open(tmp_filename, std::ios::binary | std::ios_base::out);
+    } else {
+      os_.open(tmp_filename, std::ios_base::out);
+    }
+  } catch (std::ios_base::failure& e) {
+    std::string error = e.what();
+    throw std::ios_base::failure(error + " (failed to open '" + tmp_filename
+                                 + "')");
+  }
+}
+
+StreamHandler::~StreamHandler()
+{
+  if (os_.is_open()) {
+    os_.close();
+  }
+  std::string tmp_filename = filename_ + ".tmp";
+  fs::rename(tmp_filename,
+             filename_);  // If filename_ exists it will be overwritten
```

Comment:
move comment to line before statement instead of using end of line comments. Better with git-changes and I've seen that style of commenting more in OpenROAD code.